### PR TITLE
[SPARK-10571][Core] Fix incorrect appId in Spark REST / JSON API

### PR DESCRIPTION
**Pull request by viirya**
JIRA: https://issues.apache.org/jira/browse/SPARK-10571

The app id shown in Spark REST / JSON API is not correct. This patch fixes it.

### CODE REVIEW

**Comment by JoshRosen** [comment]
I didn't highlight this super clearly in my JIRA report, but AFIAK the scope of this issue also extends to the URL parameters also mixing these up. We'll have to add new tests and backwards-compatibility rules in order to fix this without creating a new `v2` API.

**Comment by viirya** [comment]
@JoshRosen OK. Looks like this patch now only solves part of the problem. Would you like to point out where it is the URL parameters which mixes these up? I can find that myself if you are not available. Thanks.

File: core/src/main/scala/org/apache/spark/status/api/v1/OneApplicationResource.scala [hunk 1]
```diff
@@ -24,7 +24,9 @@ private[v1] class OneApplicationResource(uiRoot: UIRoot) {
 
   @GET
   def getApp(@PathParam("appId") appId: String): ApplicationInfo = {
-    val apps = uiRoot.getApplicationInfoList.find { _.id == appId }
+    // SPARK-10571: appId and appName are mixed up in previous release,
+    // To ensure backwards-compatibility, here we find both app.id == appId and app.name == appId
+    val apps = uiRoot.getApplicationInfoList.find { app => app.id == appId || app.name == appId }
```
Comment (JoshRosen) [inline core/src/main/scala/org/apache/spark/status/api/v1/OneApplicationResource.scala]: Yeah, this is the compatibility concern that I was worried about.

Comment (JoshRosen) [inline core/src/main/scala/org/apache/spark/status/api/v1/OneApplicationResource.scala]: This is the fix that I was going to propose, too. Good work.

**Comment by viirya** [comment]
I just found #8688 has been merged and looks it doesn't consider backwards-compatibility.

**Comment by srowen** [comment]
I think this and SPARK-10571 are duplicates of SPARK-10531; let's at least retag this, or close the PR, as that's already resolved.

**Pull request closed without merging**